tools-local/doc.tasks/GenerateMarkdownDoc.cs

```diff
@@ -90,14 +94,56 @@ public override bool Execute()
         private void WriteSectionProjecs(ITaskItem section, StreamWriter file)
         {
             string[] projectFiles = section.GetMetadata("Projects").Split(';');
+            var separator = Path.DirectorySeparatorChar;
 
-            file.WriteLine($"Projects in section: {projectFiles.Length}\n");
-                    
+            file.WriteLine($"Projects in section: {projectFiles.Length}");
+
+            //Logic below relies on pre-sorting to identify group/folder structure changes.
             Array.Sort(projectFiles, (a, b) => a.CompareTo(b));
 
+            string prevGroupName = null;
+
             for (int i = 0; i < projectFiles.Length; i++)
             {
+                string groupName = null;
+                string fileName = null;
+
                 var projectPath = projectFiles[i];
+                var relativeProjectPath = RelativizePath(projectPath);
+                var pathSplit = relativeProjectPath.Split(separator);
+
+                if (pathSplit == null || pathSplit.Length < 1)
+                {
+                    continue;
+                }
+                else if (pathSplit.Length == 2)
+                {
+                    groupName = pathSplit[0];
+                    fileName = pathSplit[1];
+                }
+                else //More than 2 subfolders will be grouped by [subfolders] => [project name].
+                {
+                    var end = pathSplit.Length - 1;
+                    fileName = pathSplit[end];
+
+                    var fileNameWithoutExtension = Path.GetFileNameWithoutExtension(fileName);
+
+                    //Ignore the last subfolder if its name matches the filename.
+                    if (pathSplit[end-1] == fileNameWithoutExtension)
+                    {
+                        --end;
+                    }
+
+                    groupName = string.Join(separator.ToString(), pathSplit.Take(end));
```

Comment:
if `pathSplit.Length == 1`, then, end = 0, so, a possible arrayIndexOutOfRange exception could occur

```suggestion
                    if (end > 0)
                    {
                        //Ignore the last subfolder if its name matches the filename.
                        if (pathSplit[end-1] == fileNameWithoutExtension)
                        {
                            --end;
                        }

                        groupName = string.Join(separator.ToString(), pathSplit.Take(end));
                    }
```

Reply:
Addressed and added an else to actually capture that single item.
Also changed == for comparing strings to .Equals with OrdinalIgnoreCase.